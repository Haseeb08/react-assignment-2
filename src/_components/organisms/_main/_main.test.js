import React from "react";
import { create } from "react-test-renderer";
import "@testing-library/jest-dom/extend-expect";
import Main from "./_main";
import { MemoryRouter } from "react-router";
import { BrowserRouter } from "react-router-dom";
import Home from "../_home/_home";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });
import { mount } from "enzyme";
import Gallery from "../_gallery/_gallery";
import GalleryAdmin from "../_gallery-admin/_gallery-admin";

describe("Main component testing", () => {
  test("Matches the snapshot", () => {
    const input = create(
      <BrowserRouter>
        <Main />
      </BrowserRouter>
    );
    expect(input.toJSON()).toMatchSnapshot();
  });
});

describe("routes using memory router", () => {
  const allImages = {
    images: [
      {
        id: "1",
        name: "React",
        url:
          "https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcRWXoLhb1QGq1B52D9rNOdWzmIdy0IVkmKk9Q&usqp=CAU",
      },
      {
        id: "2",
        name: "Angular",
        url: "https://angular.io/assets/images/logos/angular/angular.png",
      },
    ],
  };

  it("should show Home component for / router (using memory router)", () => {
    const component = mount(
      <MemoryRouter initialentries="{['/home']}">
        <Home />
      </MemoryRouter>
    );
    expect(component.find(Home)).toHaveLength(1);
  });
  it("should show Home component for /gallery router (using memory router)", () => {
    const component = mount(
      <MemoryRouter initialentries="{['/gallery']}">
        <Gallery allImages={allImages.images} />
      </MemoryRouter>
    );
    expect(component.find(Gallery)).toHaveLength(1);
  });
  it("should show Home component for /gallery-admin router (using memory router)", () => {
    const component = mount(
      <MemoryRouter initialentries="{['/gallery-admin']}">
        <GalleryAdmin allImages={allImages.images} />
      </MemoryRouter>
    );
    expect(component.find(GalleryAdmin)).toHaveLength(1);
  });
});
 