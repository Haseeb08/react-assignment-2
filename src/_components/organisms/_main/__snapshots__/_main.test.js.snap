// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component testing Matches the snapshot 1`] = `
<div
  className="makeStyles-main-1"
/>
`;
